sql/core/src/test/scala/org/apache/spark/sql/test/DataFrameReaderWriterSuite.scala

```diff
@@ -635,4 +638,13 @@ class DataFrameReaderWriterSuite extends QueryTest with SharedSQLContext with Be
       checkAnswer(spark.table("t"), Row(1, "a") :: Row(2, "b") :: Nil)
     }
   }
+
+  test("use saveAsTable to append to a data source table implementing CreatableRelationProvider") {
+    withTable("t") {
+      val provider = "org.apache.spark.sql.test.DefaultSource"
```

Comment:
The data source is defined in this file: https://github.com/apache/spark/pull/15996/files#diff-b9ddfbc9be8d83ecf100b3b8ff9610b9R48

I think it's easy to tell that it extends `CreatableRelationProvider` but not return a `InsertableRelation`